docs/src/stratospheric_reaction.md

```diff
@@ -0,0 +1,422 @@
+    k8 = 6.062e-15
+    k9 = 1.069e-11
+    k10 = 1.289e-2 * sigma
+    k11 = 1.0e-8
+
+    r1 = k1 * O2
+    r2 = k2 * O * O2
+    r3 = k3 * O3
+    r4 = k4 * O3 * O
+    r5 = k5 * O3
+    r6 = k6 * M * O1D
+    r7 = k7 * O1D * O3
+    r8 = k8 * O3 * NO
+    r9 = k9 * NO2 * O
+    r10 = k10 * NO2
+    r11 = k11 * NO * O
+
+    return @SMatrix [0.0 0.0 r5 0.0 0.0 0.0;
+            r6 r1+r10 r3 r1 0.0 0.0;
+            0.0 r2 0.0 0.0 0.0 0.0;
+            r7 r4+r9 r4+r7+r8 r3+r5 0.0 0.0;
+            0.0 0.0 0.0 0.0 0.0 r9+r10;
+            0.0 0.0 0.0 0.0 r8+r11 0.0]
+end
+
+function dest_static(u, p, t)
+    O1D, O, O3, O2, NO, NO2 = u
+
+    k2 = 8.018e-17
+    k11 = 1.0e-8
+
+    r2 = k2 * O * O2
+    r11 = k11 * NO * O
+
+    return @SVector [0.0, r11, 0.0, r2, 0.0, 0.0]
+end
+nothing #hide
+```
+In addition we also want to use a static vector to hold the initial conditions.
+```@example stratreac
+u0_static = @SVector [9.906e1, 6.624e8, 5.326e11, 1.697e16, 4e6, 1.093e9] # initial values
+prob_static = PDSProblem(prod_static, dest_static, u0_static, tspan) # create the PDS
+
+sol_static = solve(prob_static, MPRK43I(1.0, 0.5))
+
+nothing #hide
+```
+```@example stratreac
+using Plots
+
+plot(sol_static,
+    layout=(3,2),
+    xguide = "t [h]",
+    xguidefontsize = 8,
+    xticks = (range(first(tspan), last(tspan), 4), range(12.0, 84.0, 4)),
+    yguide=["O¹ᴰ" "O" "O₃" "O₂" "NO" "NO₂"],
+    tickfontsize = 7,
+    legend = :none, 
+    widen = true
+    )
+```
+
+The above implementation of the stratospheric reaction problem using `StaticArrays` can also be found in the [Example Problems](https://skopecz.github.io/PositiveIntegrators.jl/dev/api_reference/#Example-problems) as [`prob_pds_stratreac`](@ref).
+
+### Preservation of linear invariants
+As MPRK schemes do not preserve general linear invariants, especially when applied to non-conservative PDS, we compute and plot the relative errors with respect to both linear invariants to see how well these are preserved.
+
+```@example stratreac
+linear_invariant(a, u) = sum(a .* u)
+
+function relerr_lininv(a, u0, sol)
+    c = linear_invariant(a, u0)
+    return abs.(c .- (x -> linear_invariant(a, x)).(sol.u))./c
+end
+
+a1 = [1; 1; 3; 2; 1; 2] # first linear invariant
+a2 = [0; 0; 0; 0; 1; 1] # second linear invariant
+
+p1 = plot(sol_oop.t, relerr_lininv(a1, u0, sol_oop))
+p2 = plot(sol_oop.t, relerr_lininv(a2, u0, sol_oop))
+plot(p1, p2, 
+    xticks = (range(first(tspan), last(tspan), 4), range(12.0, 84.0, 4)),
+    legend = :none)
+```
+
+In contrast to MPRK schemes, Runge-Kutta and Rosenbrock methods preserve all linear invariants. We show this using the Rosenbrock scheme `Rosenbrock23` as an example, making sure that the solution is actually nonnegative.
+
+```@example stratreac
+using OrdinaryDiffEq
+
+sol_Ros23 = solve(prob_oop, Rosenbrock23(),
+             isoutofdomain = (y, p, t) -> any(x -> x < 0, y) # reject step if the current solution has a negative element
+             );
```

Comment:
```suggestion

# reject step if the current solution has a negative element
isoutofdomain(u, args...) = any(<(0), u) # p, t are additional arguments 
sol_Ros23 = solve(prob_oop, Rosenbrock23(); isoutofdomain);
```

Reply:
I'd suggest to take this even further by providing a function `isnegative` (and maybe also `isnonnegative`). If a solution is negative, I'd also like to know which components are affected and “how negative” the solution actually is.

Here's a draft. 

```julia
julia> using PositiveIntegrators, OrdinaryDiffEq

julia> function isnegative(u::AbstractVector, args...)
           return any(<(0), u)
       end
isnegative (generic function with 1 method)

julia> function isnegative(u::AbstractVector{<:AbstractVector}, args...)
           anyisnegative = any(isnegative, u)
           if anyisnegative
               components = Int64[]
               minima = eltype(first(u))[]
               # look for components with negative elements
               for i in eachindex(first(u))
                   v = getindex.(u,i)
                   if isnegative(v)
                       push!(components, i)
                       push!(minima, minimum(v))            
                   end
               end
               println("Components ", components ," contain negative elements, the respective minima are ", minima)
           end    
           return anyisnegative
       end
isnegative (generic function with 2 methods)

julia> isnonnegative(args...) = !isnegative(args...)
isnonnegative (generic function with 1 method)

julia> sol = solve(prob_pds_stratreac, Rosenbrock23());

julia> isnegative(sol.u)
Components [1, 2, 5] contain negative elements, the respective minima are [-1.5591986975545963e-23, -9.712921995482734e-9, -9.170872661434066e-9]
true

julia> isnonnegative(sol.u)
Components [1, 2, 5] contain negative elements, the respective minima are [-1.5591986975545963e-23, -9.712921995482734e-9, -9.170872661434066e-9]
false

julia> sol2 = solve(prob_pds_stratreac, Rosenbrock23(); isoutofdomain = isnegative);

julia> isnegative(sol2.u)
false

julia> isnonnegative(sol2.u)
true

```

I think we should check in all tutorials that the MPRK solutions are actually nonnegative, since this is what the package promises.

Reply:
That's a nice idea